packages/clerk-js/src/ui/primitives/Image.tsx

```diff
@@ -1,14 +1,22 @@
 import React from 'react';
 
+import { generateSrcSet } from '../../utils';
 import { PrimitiveProps, StateProps } from '../styledSystem';
 import { applyDataStateProps } from './applyDataStateProps';
 
-export type ImageProps = PrimitiveProps<'img'> & StateProps;
+export type ImageProps = PrimitiveProps<'img'> &
+  StateProps & {
+    size?: number;
+  };
 
 export const Image = React.forwardRef<HTMLImageElement, ImageProps>((props, ref) => {
+  const { src, size = 80, ...rest } = props;
+
   return (
     <img
-      {...applyDataStateProps(props)}
+      srcSet={generateSrcSet({ size, src, sizes: [1, 2] })}
+      src={generateSrcSet({ size, src, sizes: [2] })}
```

Comment:
❓ could we make `sizes` more readable. I imagine that it is being used to also include retina displays but it's not very clear to mean as a parameter.
Would it make more sense to add a boolean as `includeRetina  = true` instead of the array and hide the implementation detail of adding multiple `width` query params inside `generateSrcSet` ?

Reply:
Yes, you're right. I'll refactor it.